src/audio/aria/aria.c

```diff
@@ -99,35 +96,31 @@ void aria_process_data(struct comp_dev *dev,
 	cd->data_ptr = cir_buf_wrap(cd->data_ptr + sample_size, cd->data_addr, cd->data_end);
 }
 
-static int init_aria(struct comp_dev *dev, const struct comp_ipc_config *config,
-		     const void *spec)
+int aria_init(struct processing_module *mod)
 {
-	const struct ipc4_aria_module_cfg *aria = spec;
+	struct comp_dev *dev = mod->dev;
+	struct module_data *mod_data = &mod->priv;
+	struct ipc4_base_module_cfg *base_cfg = &mod_data->cfg.base_cfg;
+	const struct ipc4_aria_module_cfg *aria = mod_data->cfg.init_data;
 	struct aria_data *cd;
 	size_t ibs, chc, sgs, sgc, req_mem, att;
 	void *buf;
-	int ret;
 
-	dev->ipc_config = *config;
+	comp_info(dev, "aria_init()");
+
 	list_init(&dev->bsource_list);
 	list_init(&dev->bsink_list);
 
-	dcache_invalidate_region((__sparse_force void __sparse_cache *)spec, sizeof(*aria));
+	dcache_invalidate_region((__sparse_force void __sparse_cache *)mod_data->cfg.init_data,
+				 sizeof(*aria));
```

Comment:
this actually doesn't look correct. For IPC4 the invalidation is already done in `comp_new_ipc4()`, for IPC3 `.init_data` points to a buffer, allocated as cached and filled in with data in `module_load_config()`, so, by invalidating cache here we might drop data, written there?

Reply:
thanks, so better to remove this line than written?

Reply:
I think yes, it should be removed, but please make it a separate PR. Is aria used anywhere with IPC3?

Reply:
or a separate commit? No, as I know the aria only supports 32-bit container and we don't have aria before mtl.

Reply:
if it's IPC4 only, then yes, it's a smaller change and just a separate commit should be enough